.github/scripts/validateActionsAndWorkflows.sh

```diff
@@ -94,4 +100,43 @@ fi
 # Cleanup after ourselves and delete actionlint
 rm -rf ./actionlint
 
+###############################################################################
+#                      Check for unsafe action references                     #
+###############################################################################
+title 'Checking for mutable action references...'
+find . -type f \( -name "*.yml" -o -name "*.yaml" \) -print0 \
```

Comment:
This is pretty hard to read, and there's no need to be fancy with everything in a single pipeline. I recommend using a regular `for` loop instead and breaking this into pieces so you can add some comments explaining what it's doing.

Reply:
Updated this to break it up into smaller pieces with comments explaining what each part is doing.